### ISSUE TITLE
Search returns no results since migrating DocSearch

### ISSUE THREAD

**Issue by joe-bell** [facebook/docusaurus#7647]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

Prior to the DocSearch migration, [plaiceholder.co's search integration](http://plaiceholder.co/docs) worked seamlessly

Since upgrading, I've been unable to get the search up and running again for over a week now – this is my resort! I've followed steps in #4872, #6693 and the [migration guide](https://docusaurus.io/blog/2021/11/21/algolia-docsearch-migration), with no luck.

Additional context to the repro provided:

<details>
  <summary>Crawler Config</summary>

```ts
new Crawler({
  appId: "<MY-APP-ID>",
  apiKey: "<MY-API-KEY>",
  rateLimit: 8,
  startUrls: ["https://plaiceholder.co/docs", "https://plaiceholder.co/"],
  sitemaps: ["https://plaiceholder.co/docs/sitemap.xml"],
  ignoreCanonicalTo: true,
  discoveryPatterns: ["https://plaiceholder.co/**"],
  actions: [
    {
      indexName: "plaiceholder",
      pathsToMatch: ["https://plaiceholder.co/**"],
      recordExtractor: ({ $, helpers }) => {
        // priority order: deepest active sub list header -> navbar active item -> 'Documentation'
        const lvl0 =
          $(
            ".menu__link.menu__link--sublist.menu__link--active, .navbar__item.navbar__link--active"
          )
            .last()
            .text() || "Documentation";

        return helpers.docsearch({
          recordProps: {
            lvl0: {
              selectors: "",
              defaultValue: lvl0,
            },
            lvl1: ["header h1", "article h1"],
            lvl2: "article h2",
            lvl3: "article h3",
            lvl4: "article h4",
            lvl5: "article h5, article td:first-child",
            lvl6: "article h6",
            content: "article p, article li, article td:last-child",
          },
          indexHeadings: true,
          aggregateContent: true,
          recordVersion: "v3",
        });
      },
    },
  ],
  initialIndexSettings: {
    plaiceholder: {
      attributesForFaceting: [
        "type",
        "lang",
        "language",
        "version",
        "docusaurus_tag",
      ],
      attributesToRetrieve: [
        "hierarchy",
        "content",
        "anchor",
        "url",
        "url_without_anchor",
        "type",
      ],
      attributesToHighlight: ["hierarchy", "content"],
      attributesToSnippet: ["content:10"],
      camelCaseAttributes: ["hierarchy", "content"],
      searchableAttributes: [
        "unordered(hierarchy.lvl0)",
        "unordered(hierarchy.lvl1)",
        "unordered(hierarchy.lvl2)",
        "unordered(hierarchy.lvl3)",
        "unordered(hierarchy.lvl4)",
        "unordered(hierarchy.lvl5)",
        "unordered(hierarchy.lvl6)",
        "content",
      ],
      distinct: true,
      attributeForDistinct: "url",
      customRanking: [
        "desc(weight.pageRank)",
        "desc(weight.level)",
        "asc(weight.position)",
      ],
      ranking: [
        "words",
        "filters",
        "typo",
        "attribute",
        "proximity",
        "exact",
        "custom",
      ],
      highlightPreTag: '<span class="algolia-docsearch-suggestion--highlight">',
      highlightPostTag: "</span>",
      minWordSizefor1Typo: 3,
      minWordSizefor2Typos: 7,
      allowTyposOnNumericTokens: false,
      minProximity: 1,
      ignorePlurals: true,
      advancedSyntax: true,
      attributeCriteriaComputedByMinProximity: true,
      removeWordsIfNoResults: "allOptional",
      separatorsToIndex: "_",
    },
  },
});
```

</details>

---

I love using Docusaurus, so I'd really appreciate any support to get this up and running again

### Reproducible demo

https://github.com/joe-bell/plaiceholder/tree/main/docs

### Steps to reproduce

1. Visit https://plaiceholder.co/docs
2. Click on the search bar, or visit `/docs/search`
3. Enter `get` (this did return results prior to upgrade)
4. Enter any combination of letters or numbers
5. See that no results are returned

### Expected behavior

In Algolia's Crawler editor page, I'm able to see that this should return results:

<img width="1293" alt="CleanShot 2022-06-18 at 09 36 29@2x" src="https://user-images.githubusercontent.com/7349341/174448188-491bf244-271c-4b64-9992-0fa84ca8e530.png">

### Actual behavior

Typing `get` (or any search) returns no results

<img width="1798" alt="CleanShot 2022-06-18 at 09 32 27@2x" src="https://user-images.githubusercontent.com/7349341/174448047-677194d8-3fa9-4b00-91f2-dd5a55529653.png">

### Your environment

- Public source code: https://github.com/joe-bell/plaiceholder/tree/main/docs
- Public site URL: https://plaiceholder.co/docs
- Docusaurus version used: 2.0.0-beta.21
- Environment name and version (e.g. Chrome 89, Node.js 16.4): N/A (all affected)
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): N/A (all affected)

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by Josh-Cena** [comment]
cc @shortcuts

**Comment by shortcuts** [comment]
Hey, looking at the requests sent, `contextualSearch` option is set to `true` on the Docusaurus side but your Crawler config has changed (you can see the warning message on 1st screenshot) to a Docusaurus v2 config

This is indeed related to the issue you've linked, you can follow both https://github.com/facebook/docusaurus/issues/6693#issuecomment-1158568333 or https://github.com/facebook/docusaurus/issues/6693#issuecomment-1158639529 for the solution.

TLDR: Delete Algolia index from the dashboard and start a new crawl

**Comment by joe-bell** [comment]
Wow, a classic case of "turn it off and on again" 🙈 

Thanks so much @shortcuts, that resolved it for me

**Comment by Josh-Cena** [comment]
Thanks @shortcuts!

**Comment by slorber** [comment]
## No search result?

For anyone passing by, if you don't get any Algolia search results:

- make sure that your Algolia index has the fields in the screenshot below
- If you don't see these fields, then you have an index config problem
- You should check your crawler config, make sure it matches the recommended one, and then delete/recreate your index based on the newly updated/fixed crawler config ([Algolia team recommendation](https://github.com/facebook/docusaurus/issues/9200#issuecomment-1667338492))

![image](https://github.com/facebook/docusaurus/assets/749374/b518cfe4-6b0c-480a-ab7b-be9f098f9cd3)

See also: https://github.com/facebook/docusaurus/discussions/10007#discussioncomment-9021352
